### ISSUE TITLE
Feature request: allow comment style to specify font and size.

### ISSUE THREAD

**Issue by gfbee** [racket/drracket#299]

**Comment by spdegabrielle** [comment]
TL;DR close issue as ‘not DrRacket’ - should be a plug-in and a `#lang`.  Would support as a plug-in project.

——

Hi

I understand the motivation for this issue but the comments in the current `#lang`’s supported by DrRacket _rely_ on monospace fonts for formatting existing comments. 

As the GUI library supports proportional fonts and is distinct from DrRacket, and this is not an issue that would be resolved _in_ DrRacket code I think this should be closed. 

That said, I think this is an idea worthy of investigation. An implementation would probably require a custom `#lang` that is the same as normal racket, and a DrRacket plug-in that recognises the custom `#lang` and supports proportional comments. 

If you were to embark on this project I believe community members would be supportive.  I’d be interested in proportional fonts for code (using tabs for indenting), or even code and comments with implicit and explicit font styles. I’d be happy to discuss on racket-users or on the racket slack.

Kind regards
Stephen

**Comment by gfbee** [comment]
I'm not proposing this for expression comments, which have horizontal structure. The issue you raise is really whether other code comments (end-of-line or block) imply horizontal structure. If they do, then we don't have a prose comment form. One route is to have more kinds of comment, and then the `#lang` and tool mechanisms can be used to react to the semantics.

But you can choose a variable-width font right now in DrRacket, and ruin horizontal layout of both code and comments. If there's a file with a lot of prose commentary that could be worth it. But unless you're saying that source requires all-fixed or all-variable, then it must be fine to have a mix. In any case, I would find it very useful to be able to choose for myself whether to view the comments in a fixed or variable font. I've already hard-coded this into the csc104 language today, and it makes many of the course files look much nicer, and gets me a large amount of the benefit that notebooks have.

<img width="679" alt="commentary-and-code" src="https://user-images.githubusercontent.com/4381135/63183121-327d3c00-c022-11e9-9e28-2875959dcf60.png">

**Comment by spdegabrielle** [comment]
Thanks @gfbee that looks great! You are correct that the guy toolkit supports a mix of proportional and monospace in a variety of fonts

If you are happy yo put your code on your GitHub we can continue the conversation there.

Kind regards,
Stephen
